### PR TITLE
Add MaskedBehavior

### PR DESCRIPTION
**Pull request by bijington**
Documentation added for .NET MAUI Community Toolkit proposals:

- [Proposal] `MaskedBehavior` <https://github.com/CommunityToolkit/Maui/issues/46>

### CODE REVIEW

**Comment by PRMerger12** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Index File | This PR contains a change to the index file.

File: docs/maui/behaviors/masked-behavior.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    public MaskedBehaviorPage()
+    {
+        Content = new Entry
+        {
+            Keyboard = Keyboard.Numeric
+        }.Behaviors(
+            new MaskedBehavior
+            {
+                Mask = "XXXX XXXX XXXX XXXX"
+            });
```
Comment (bijington) [inline docs/maui/behaviors/masked-behavior.md:82]:
```suggestion
        }.Behaviors(new MaskedBehavior
        {
            Mask = "XXXX XXXX XXXX XXXX"
        });
```

**Comment by PRMerger18** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Index File | This PR contains a change to the index file.

File: docs/maui/behaviors/index.md [hunk 2]
```diff
@@ -19,5 +19,6 @@ The .NET MAUI Community Toolkit provides a collection of pre-built, reusable beh
 
 | Converter | Description |
 | --------- | ----------- |
+| [`MaskedBehavior`](masked-behavior.md) | The `MaskedBehavior` is a `Behavior` that allows the user to define an input mask for data entry. |
```
Comment (jfversluis) [inline docs/maui/behaviors/index.md:22]: We don't `Behavior` the rest in this table as well. I'm fine with either, but I think we should make it consistent at least :)

File: docs/maui/behaviors/masked-behavior.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        entry.Behaviors.Add(behavior);
+
+        Content = entry;
+    }
+}
+```
+
+### C# Markup
+
+Our [`CommunityToolkit.Maui.Markup`](../markup/markup.md) package provides a much more concise way to use this converter in C#.
```
Comment (jfversluis) [inline docs/maui/behaviors/masked-behavior.md:66]:
```suggestion
Our [`CommunityToolkit.Maui.Markup`](../markup/markup.md) package provides a much more concise way to use this behavior in C#.
```

File: docs/maui/behaviors/masked-behavior.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+}
+```
+
+## Properties
+
+|Property  |Type  |Description  |
+|---------|---------|---------|
+| `Mask` | `string` | The mask that the input value needs to match. |
+| `UnmaskedCharacter` | `char` | The placeholder character for when no input has been given yet. |
```
Comment (jfversluis) [inline docs/maui/behaviors/masked-behavior.md:91]: "The placeholder character for when no input has been given yet." Is this description right? If it is, it's not immediately clear to me what this is supposed to do

**Review by jfversluis — CHANGES_REQUESTED** [review]

**Comment by PRMerger10** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Index File | This PR contains a change to the index file.

File: docs/maui/behaviors/masked-behavior.md (hunk 3 above)
Comment (bijington) [reply docs/maui/behaviors/masked-behavior.md:66]: I need to stop these! 🤦

**Comment by PRMerger12** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Index File | This PR contains a change to the index file.

File: docs/maui/behaviors/masked-behavior.md (hunk 4 above)
Comment (bijington) [reply docs/maui/behaviors/masked-behavior.md:91]: I took this from the xml documentation. I believe it means that if we have a mask of `XX XX` with an `UnmaskedCharacter` of `_` and the user hasn't entered anything then the Entry would show `__ __`. I'll have a play and see what I can come up with

**Comment by PRMerger16** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.

File: docs/maui/behaviors/masked-behavior.md (hunk 4 above)
Comment (bijington) [reply docs/maui/behaviors/masked-behavior.md]: Let me know your thoughts on the change

**Comment by PRMerger9** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.

**Comment by PRMerger20** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.

**Review by jfversluis — APPROVED** [review]

**Comment by PRMerger-2** [comment]
## PRMerger Results
 | Issue | Description |
 --- | --- |
 Added File(s) | This PR contains added files. New files require human review.
 Yaml File(s) | This PR includes changes to .yml file(s) owned by another author.
 File Change Percent | This PR contains file(s) with more than 20% file change.
 Image File | This PR added or updated an image file(s).
 Index File | This PR contains a change to the index file.

**Pull request merged**